time_tracker_api/activities/activities_model.py

```diff
@@ -39,12 +46,64 @@ def __str___(self):
         return "the activity \"%s\"" % self.name  # pragma: no cover
 
 
-def create_dao() -> ActivityDao:
-    repository = CosmosDBRepository.from_definition(container_definition,
-                                                    mapper=ActivityCosmosDBModel)
+class ActivityCosmosDBRepository(CosmosDBRepository):
+    def __init__(self):
+        CosmosDBRepository.__init__(
+            self,
+            container_id=container_definition['id'],
+            partition_key_attribute='tenant_id',
+            mapper=ActivityCosmosDBModel,
+        )
+
+    def create_sql_in_condition(self, activities_ids_list):
+        id_values = convert_list_to_tuple_string(activities_ids_list)
+
+        return "c.id IN {value_condition}".format(value_condition=id_values)
+
+    def find_all_with_id_in_list(
+        self,
+        event_context: EventContext,
+        activities_ids_list: List[str],
+        visible_only=True,
+        mapper: Callable = None,
+    ):
+        visibility = self.create_sql_condition_for_visibility(visible_only)
+        query_str = """
+            SELECT * FROM c
+            WHERE {condition}
+            {visibility_condition}
+            """.format(
+            condition=self.create_sql_in_condition(activities_ids_list),
+            visibility_condition=visibility,
+        )
+
+        tenant_id_value = self.find_partition_key_value(event_context)
+        result = self.container.query_items(
+            query=query_str,
+            partition_key=tenant_id_value,
+        )
+
+        function_mapper = self.get_mapper_or_dict(mapper)
+        return list(map(function_mapper, result))
 
-    class ActivityCosmosDBDao(APICosmosDBDao, ActivityDao):
-        def __init__(self):
-            CosmosDBDao.__init__(self, repository)
 
-    return ActivityCosmosDBDao()
+class ActivityCosmosDBDao(APICosmosDBDao, ActivityDao):
+    def __init__(self, repository):
+        CosmosDBDao.__init__(self, repository)
+
+    def get_all_with_id_in_list(
+        self,
+        activities_ids_list,
+    ):
+        event_ctx = self.create_event_context("read-many")
+        activities_list = self.repository.find_all_with_id_in_list(
```

Comment:
We could return the value directly.

Reply:
Yeah, but I think that it is more legible in that way.